src/transition/CMakeLists.txt

```diff
@@ -18,48 +18,70 @@
 # This file creates the basic target(s) needed by all backends
 #
 
-assert_defined_variables( WITH_REFERENCE_BACKEND WITH_OMP_BACKEND )
+assert_defined_variables( WITH_REFERENCE_BACKEND WITH_OMP_BACKEND WITH_NONBLOCKING_BACKEND )
 
-if( WITH_REFERENCE_BACKEND )
-	add_library( sparseblas_static STATIC
-		${CMAKE_CURRENT_SOURCE_DIR}/sparseblas.cpp
+function( add_transition_library target_name lib_type lib_name src1 )
+
+	set( multiValueArgs
+		"SOURCES"
+		"PUBLIC_LINK_LIBRARIES"
 	)
+	cmake_parse_arguments( parsed "" "" "${multiValueArgs}" "SOURCES;${src1};${ARGN}" )
 
-	set_target_properties( sparseblas_static PROPERTIES
-		OUTPUT_NAME "sparseblas"
+	add_library( ${target_name} ${lib_type} ${parsed_SOURCES} )
+	set_target_properties( ${target_name} PROPERTIES
+		OUTPUT_NAME ${lib_name}
 		ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/shmem"
 	)
 
-	target_link_libraries( sparseblas_static PUBLIC backend_reference transition )
+	target_link_libraries( ${target_name} PUBLIC transition_headers ${parsed_PUBLIC_LINK_LIBRARIES} )
+	target_link_libraries( ${target_name} PRIVATE backend_flags )
+	add_dependencies( libs ${target_name} )
+	install( TARGETS ${target_name} EXPORT GraphBLASTargets
+		ARCHIVE DESTINATION "${SHMEM_BACKEND_INSTALL_DIR}"
+	)
 
-	target_link_libraries( sparseblas_static PRIVATE backend_flags )
+endfunction( add_transition_library )
 
-	add_dependencies( libs sparseblas_static )
+if( WITH_REFERENCE_BACKEND )
+	add_transition_library( sparseblas_static STATIC "sparseblas" ${CMAKE_CURRENT_SOURCE_DIR}/sparseblas.cpp
+		PUBLIC_LINK_LIBRARIES backend_reference
+	)
 
-	install( TARGETS sparseblas_static
-		EXPORT GraphBLASTargets
-		ARCHIVE DESTINATION "${SHMEM_BACKEND_INSTALL_DIR}"
+	set( _SPBLAS_PREFIX "kml_" )
+	if( SPBLAS_PREFIX )
+		set( _SPBLAS_PREFIX "${SPBLAS_PREFIX}_" )
+	endif()
+	add_transition_library( ${_SPBLAS_PREFIX}sparseblas_static STATIC "${_SPBLAS_PREFIX}sparseblas"
```

Comment:
Should the name not be something like `sparseblas_sequential` rather than "sparseblas"?

Reply:
fixed